sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/signaling/CommunicationSignalingClient.java

```diff
@@ -0,0 +1,148 @@
+package com.azure.android.communication.chat.signaling;
+
+import android.content.Context;
+
+import com.azure.android.communication.chat.signaling.properties.ChatEventId;
+import com.microsoft.trouterclient.ISelfHostedTrouterClient;
+import com.microsoft.trouterclient.ITrouterAuthHeadersProvider;
+import com.microsoft.trouterclient.ITrouterConnectionDataCache;
+import com.microsoft.trouterclient.TrouterClientHost;
+import com.microsoft.trouterclient.registration.ISkypetokenProvider;
+import com.microsoft.trouterclient.registration.TrouterSkypetokenAuthHeaderProvider;
+import com.microsoft.trouterclient.registration.TrouterUrlRegistrar;
+import com.microsoft.trouterclient.registration.TrouterUrlRegistrationData;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.azure.android.communication.chat.signaling.SignalingConfig.defaultRegistrarHostnameAndBasePath;
+import static com.azure.android.communication.chat.signaling.SignalingConfig.defaultTrouterHostname;
+import static com.azure.android.communication.chat.signaling.SignalingConfig.platform;
+import static com.azure.android.communication.chat.signaling.SignalingConfig.platformUiVersion;
+import static com.azure.android.communication.chat.signaling.SignalingConfig.testApplicationId;
+import static com.azure.android.communication.chat.signaling.SignalingConfig.testTemplateKey;
+import static com.azure.android.communication.chat.signaling.SignalingConfig.trouterClientVersion;
+
+public class CommunicationSignalingClient implements SignalingClient {
+    private TrouterClientHost trouterClientHost;
+    private ISelfHostedTrouterClient trouter;
+    String userToken;
+    Set<CommunicationListener> trouterListeners;
```

Comment:
are we accessing `trouterListeners` and `userToken` outside of this class but within the package? just making sure its package private visibility is intentional